kong/plugins/acme/storage/config_adapters/redis.lua

```diff
@@ -7,6 +7,8 @@ local function redis_config_adapter(conf)
         ssl = conf.ssl,
         ssl_verify = conf.ssl_verify,
         ssl_server_name = conf.server_name,
+        username = conf.username,
```

Comment:
these two will need compat code. a warning to be throwed at CP side would be sufficient.

Reply:
I don't think compat code is required here 🤔 . This is not a schema change but library internals change. CP and DP config stays the same. DP will connect to Redis the same way it used to and until it's upgraded it'll be suffering from the same bug but I don't think that's fixable by compat code. I might be wrong though 🤷

Reply:
I tested CP@master and DP@3.6.1 and didn't have any compat issues.

Created the plugin:
```
http :8001/plugins name=acme config.storage=redis config.storage_config.redis.host=localhost config.storage_config.redis.username=default config.storage_config.redis.password=passdefault config.account_email=test@konghq.com -f
```

DP logs:
```
kong-dp  | 127.0.0.1 - - [14/Aug/2024:13:10:58 +0000] "GET /A HTTP/1.1" 200 360 "-" "Wget/1.21.2" kong_request_id: "c0de772ceafaf3397e04510004ff0e5a"
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:47 acme: config.storage_config.redis.auth is deprecated, please use config.storage_config.redis.password instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:56 acme: config.storage_config.redis.ssl_server_name is deprecated, please use config.storage_config.redis.server_name instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:66 acme: config.storage_config.redis.namespace is deprecated, please use config.storage_config.redis.extra_options.namespace instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:75 acme: config.storage_config.redis.scan_count is deprecated, please use config.storage_config.redis.extra_options.scan_count instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:47 acme: config.storage_config.redis.auth is deprecated, please use config.storage_config.redis.password instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:56 acme: config.storage_config.redis.ssl_server_name is deprecated, please use config.storage_config.redis.server_name instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:66 acme: config.storage_config.redis.namespace is deprecated, please use config.storage_config.redis.extra_options.namespace instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:75 acme: config.storage_config.redis.scan_count is deprecated, please use config.storage_config.redis.extra_options.scan_count instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [notice] 626#0: *7 [lua] init.lua:259: purge(): [DB cache] purging (local) cache, context: ngx.timer
```

CP logs:

```
kong-dp  | 127.0.0.1 - - [14/Aug/2024:13:10:58 +0000] "GET /A HTTP/1.1" 200 360 "-" "Wget/1.21.2" kong_request_id: "c0de772ceafaf3397e04510004ff0e5a"
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:47 acme: config.storage_config.redis.auth is deprecated, please use config.storage_config.redis.password instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:56 acme: config.storage_config.redis.ssl_server_name is deprecated, please use config.storage_config.redis.server_name instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:66 acme: config.storage_config.redis.namespace is deprecated, please use config.storage_config.redis.extra_options.namespace instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:75 acme: config.storage_config.redis.scan_count is deprecated, please use config.storage_config.redis.extra_options.scan_count instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:47 acme: config.storage_config.redis.auth is deprecated, please use config.storage_config.redis.password instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:56 acme: config.storage_config.redis.ssl_server_name is deprecated, please use config.storage_config.redis.server_name instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:66 acme: config.storage_config.redis.namespace is deprecated, please use config.storage_config.redis.extra_options.namespace instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [warn] 626#0: *7 [kong] schema.lua:75 acme: config.storage_config.redis.scan_count is deprecated, please use config.storage_config.redis.extra_options.scan_count instead (deprecated after 4.0), context: ngx.timer
kong-dp  | 2024/08/14 13:10:57 [notice] 626#0: *7 [lua] init.lua:259: purge(): [DB cache] purging (local) cache, context: ngx.timer
```

Reply:
This is not an issue that stop DP from starting I agree. The problem is that user that configured such options on 3.8 needs
to get notified that, although the schema validation passed, the two options does nothing on 3.7 or older DP.
Thus I mentioned a warning log would be enough, similar as we did for other ones in https://github.com/Kong/kong/blob/master/kong/clustering/compat/checkers.lua

Reply:
Done in https://github.com/Kong/kong/pull/13496/commits/58dc72f78c4b0b1e100e5c2878878efad1c00347:

Result:
```
kong-cp  | 2024/08/15 16:49:36 [warn] 834#0: *1293 [lua] checkers.lua:20: log_warn_message(): [clustering] Kong Gateway v3.8.0 configures acme plugin with redis username which is incompatible with dataplane version 3.6.1 and will will be removed.. [id: b63a259b-dba5-4da8-8938-e16db77f8584, host: 7f3900754d3e, ip: 172.18.0.7, version: 3.6.1], client: 172.18.0.7, server: kong_cluster_listener, request: "GET /v1/outlet?node_id=b63a259b-dba5-4da8-8938-e16db77f8584&node_hostname=7f3900754d3e&node_version=3.6.1 HTTP/1.1", host: "kong-cp:8005"
kong-cp  | 2024/08/15 16:49:36 [warn] 834#0: *1293 [lua] checkers.lua:20: log_warn_message(): [clustering] Kong Gateway v3.8.0 configures acme plugin with redis password which is incompatible with dataplane version 3.6.1 and will will be removed.. [id: b63a259b-dba5-4da8-8938-e16db77f8584, host: 7f3900754d3e, ip: 172.18.0.7, version: 3.6.1], client: 172.18.0.7, server: kong_cluster_listener, request: "GET /v1/outlet?node_id=b63a259b-dba5-4da8-8938-e16db77f8584&node_hostname=7f3900754d3e&node_version=3.6.1 HTTP/1.1", host: "kong-cp:8005"
```